### PR TITLE
fix: correct detection of obsolete files

### LINKED ISSUE #775: Detecting / deleting obsolete files not working
**Describe the bug**
Delete obsolete is not detecting deleted / non-existing file.

**To Reproduce**
1. Configuration file: 

```yml
preserve_hierarchy: true

files:
  # Docs Markdown files
  - source: '/test/en/**/*.md'
    translation: '/test/%two_letters_code%/**/%original_file_name%'
```

2. `tree test/`:

```bash
test/
└── en
    ├── support.md
    └── test.md

2 directories, 2 files
```

3. Execute CLI command: `npx @crowdin/cli upload sources --token ... --project-id ... --delete-obsolete --no-auto-update`
Outputs:

```bash
✔️  Fetching project info     
⏭  File 'test/en/support.md'
⏭  File 'test/en/test.md'
✔️  No obsolete files were found
✔️  No obsolete directories found
```

image

5. Rename file `test/en/support.md` => `test/en/help.md`
6.  Execute CLI command again: `npx @crowdin/cli upload sources --token ... --project-id ... --delete-obsolete --no-auto-update`
Outputs:

```bash
✔️  Fetching project info     
⏭  File 'test/en/test.md'
⏭  File 'test/en/help.md'
✔️  No obsolete files were found
✔️  No obsolete directories found
```

image

**Expected behavior**
I expect the file `test/en/support.md` to be detected as obsolete and deleted on Crowdin afterwards as it does not exists locally anymore.

**Environment:**
 - OS: Ubuntu
 - Version 23.04
 - Cli: @crowdin/cli@3.19.2

### PR DESCRIPTION
**Pull request by anbraten**
closes #775 

Repro: https://github.com/anbraten/repro-crowdin-delete-obsolete

I am not sure why the `exportPattern` of a source-file was checked to see that it is obsolete. It works for me without the check, but I would also be happy to fix the check if it's needed and someone could explain me how it should work.

### CODE REVIEW

File: src/test/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        boolean preserveHierarchy = true;
+        List<String> filesToUpload = Arrays.asList("test/en/test.md", "test/en/help.md");
+        String pattern = "/test/en/*.md";
+        String exportPattern = "/test/%two_letters_code%/%original_path%/%original_file_name%";
+        List<String> ignorePattern = Arrays.asList("**/.*");
+
+        Map<String, File> obsoleteFiles = ObsoleteSourcesUtils.findObsoleteProjectFiles(projectFiles, preserveHierarchy,
+                filesToUpload, pattern, exportPattern, ignorePattern);
+
+        assertEquals(1, obsoleteFiles.size());
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtilsTest.java:37]: I would suggest being stricter here and comparing not only the number of files, but the exact file path.

Comment (anbraten) [reply src/test/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtilsTest.java:37]: added an explicit check for the file as well.

**Comment by anbraten** [comment]
Not sure where the windows issue was exactly coming from. Probably some path separator mismatch.

**Comment by andrii-bodnar** [comment]
@anbraten yes, most likely it is related to paths. We already have some similar tests, and the easiest option here is to have separate test cases for Linux/Mac and Windows operating systems.

These tests have the following annotations:

```java
@DisabledOnOs({OS.LINUX, OS.MAC})
@DisabledOnOs(OS.WINDOWS)
```

Or we can somehow normalize the paths to unify them for those environments.

File: src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java [hunk 2]
```diff
@@ -29,7 +29,10 @@ public static Map<String, File> findObsoleteProjectFiles(
 
     private static boolean checkExportPattern(String exportPattern, File file) {
         String fileExportPattern = ProjectFilesUtils.getExportPattern(file.getExportOptions());
-        return exportPattern.equals(fileExportPattern != null ? Utils.normalizePath(fileExportPattern) : null);
+        if (fileExportPattern == null) {
```
Comment (anbraten) [inline src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java:33]: @andrii-bodnar What do you think of this approach?

Comment (andrii-bodnar) [reply src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java:33]: Does this mean that files with the `null` export pattern will be considered obsolete and deleted, or the opposite?

Comment (anbraten) [reply src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java:33]: It basically uses all projectFiles (files received by the api) and checks for each file:
- if that file matches the config patterns (source-pattern, ignore and preserveHierarchy)
- if it matches the exportPattern (if the api returns null for the export pattern we ignore it now by this change)
- and if the file has `isFileDontHaveUpdate=true` (which seems to check if the file is not part of the local files anymore)

Comment (anbraten) [reply src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java:33]: I adjusted some var and method names so they match the rest of the code and are hopefully more descriptive.

Comment (andrii-bodnar) [reply src/main/java/com/crowdin/cli/commands/functionality/ObsoleteSourcesUtils.java:33]: @anbraten great, I just tested it locally and the fix works for me, thank you! 🚀

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**